packages/react-components/analytics/CeloAnalytics.tsx

```diff
@@ -82,26 +87,70 @@ class CeloAnalytics {
 
   track(eventName: string, eventProperties: {}, attachDeviceInfo = false) {
     if (!this.isEnabled()) {
-      this.Logger.info(
-        'CeloAnalytics/track',
-        `Analytics is disabled, not tracking event ${eventName}`
-      )
+      this.Logger.info(TAG, `Analytics is disabled, not tracking event ${eventName}`)
       return
     }
+
+    this.Logger.info(TAG, `Tracking event ${eventName}`, JSON.stringify(eventProperties))
+
     if (!SEGMENT_API_KEY) {
       return
     }
 
     try {
-      this.Logger.info('CeloAnalytics/track', `Attempting to tracking event ${eventName}`)
       const props = this.getProps(eventProperties)
       if (attachDeviceInfo) {
         _.set(props, 'device', getDeviceInfo())
       }
       Analytics.track(eventName, props)
     } catch (err) {
-      this.Logger.error('CeloAnalytics/track', `Failed to tracking event ${eventName}`, err)
+      this.Logger.error(TAG, `Failed to tracking event ${eventName}`, err)
+    }
+  }
+
+  // Used with trackSubEvent and endTracking to track durations for
+  // processes with multiple steps. For one-off events, use track method
+  startTracking(eventName: string) {
+    this.activeEvents.set(
+      eventName,
+      new Map<string, number>([['__startTrackingTime__', Date.now()]])
+    )
+  }
+
+  // See startTracking
+  trackSubEvent(eventName: string, subEventName: string) {
+    if (!this.activeEvents.has(eventName)) {
+      return this.Logger.warn(TAG, 'Attempted to track sub event for invalid event. Ignoring.')
+    }
+
+    this.activeEvents.get(eventName)!.set(subEventName, Date.now())
+  }
+
+  // See startTracking
+  stopTracking(eventName: string, eventProperties = {}) {
+    if (!this.activeEvents.has(eventName)) {
+      return
+    }
+
+    const subEvents = this.activeEvents.get(eventName)!
+    if (subEvents.size === 1) {
+      return this.Logger.warn(TAG, 'stopTracking called for event without subEvents. Ignoring.')
     }
+
+    const durations: { [subEventName: string]: number } = {}
+    let prevEventTime = subEvents.get('__startTrackingTime__')!
+    for (const [subEventName, timestamp] of subEvents) {
+      if (subEventName === '__startTrackingTime__') {
+        continue
+      }
+      durations[subEventName] = timestamp - prevEventTime
```

Comment:
How does this logic work with parallel events like reveals or verify. If I understand correctly, we will record the maximum time of each event? It would be useful to have access to the individual timing of each reveal/verify

Reply:
Correct. Just a limitation of this simple design. We _could_ register separate events and subevents to track that 3 reveal/complete txs but seemed like overkill